src/mdns/mdns_avahi.cpp

```diff
@@ -616,9 +679,8 @@ otbrError PublisherAvahi::PublishService(const char *       aHostName,
     std::string        fullHostName;
     // aligned with AvahiStringList
     AvahiStringList  buffer[(kMaxSizeOfTxtRecord - 1) / sizeof(AvahiStringList) + 1];
-    AvahiStringList *last = nullptr;
-    AvahiStringList *curr = buffer;
-    size_t           used = 0;
+    AvahiStringList *head = nullptr;
+    SuccessOrExit(error = TxtListToAvahiStringList(aTxtList, buffer, sizeof(buffer), head));
```

Comment:
nit: empty line after declaration block
```suggestion
    AvahiStringList *head = nullptr;

    SuccessOrExit(error = TxtListToAvahiStringList(aTxtList, buffer, sizeof(buffer), head));
```